### PR TITLE
[New Model] HiFiGAN

### PR DESCRIPTION
**Pull request by erogol**
This PR introduces the initial implementation of HiFIGAN in :frog: territory. 

- It is mainly adopted from the official implementation https://github.com/jik876/hifi-gan
- It also enables us to import officially released models with slight on-the-fly layer restructuring.
- audio.py is also updated to allow different ```log ``` scales as it converts amplitude specs to DB.  Traditionally we use ```log10``` however some of the other projects (including HifiGAN) use ```log``` scaling. This update allows us to train compatible models with the other projects. 
- Different than the original abstractive design of frog: TTS, we adopt HiFiGAN models with self-sufficient generator and discriminator model files to experiment with the pros and cons of this design choice.
- Optimizers are configurable from config.json file.
- Scheduler update steps are moved to the end of the epoch loop.

Thx :crown: @rishikksh20 who initiated the work and helped a lot to experiment and find the bugs.

### CODE REVIEW

File: TTS/vocoder/configs/modified_hifigan.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    // TRAINING
+    "batch_size": 16,       // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.
+
+    // VALIDATION
+    "run_eval": true,
+    "test_delay_epochs": 10,  //Until attention is aligned, testing only wastes computation time.
+    "test_sentences_file": null,  // set a file to load sentences to be used for testing. If it is null then we use default english sentences.
+
+    // OPTIMIZER
```
Comment (rishikksh20) [inline TTS/vocoder/configs/modified_hifigan.json:125]: Doesn't `optimizer_gen = getattr(torch.optim, c.optimizer)` required `optimizer` key in config file or it load any default optimizer?

Comment (erogol) [reply TTS/vocoder/configs/modified_hifigan.json]: it should be fixed with the new commits.
